azext_iot/iothub/providers/message_endpoint.py

```diff
@@ -224,6 +219,174 @@ def create(
+        encoding: Optional[str] = None,
+        batch_frequency: Optional[int] = None,
+        chunk_size_window: Optional[int] = None,
+        file_name_format: Optional[str] = None,
+        endpoint_uri: Optional[str] = None,
+        entity_path: Optional[str] = None,
+        database_name: Optional[str] = None,
+        primary_key: Optional[str] = None,
+        secondary_key: Optional[str] = None,
+        partition_key_name: Optional[str] = None,
+        partition_key_template: Optional[str] = None,
+        identity: Optional[str] = None
+    ):
+        # if nothing is provided -> should we block?
+        # have the user say the type. Will make args easier (as in we do not need to check for unneeded args)
+        original_endpoint = self._show_by_type(endpoint_name=endpoint_name, endpoint_type=endpoint_type)
+
+        if any([connection_string, primary_key, secondary_key]) and identity:
+            cosmos_db = endpoint_type.lower() == EndpointType.CosmosDBContainer.value
+            optional_msg = ", --primary-key and/or --secondary-key," if cosmos_db else ""
+            error_msg = "Please use either --connection-string" + optional_msg + " or --identity."
+            raise MutuallyExclusiveArgumentError(error_msg)
+
+        # Properties for all endpoint types
+        if endpoint_resource_group:
+            original_endpoint.resource_group = endpoint_resource_group
+        if endpoint_subscription_id:
+            original_endpoint.subscription_id = endpoint_subscription_id
+        if endpoint_uri:
+            # Handle this later with cosmos db connection string parsing
+            original_endpoint.endpoint_uri = endpoint_uri
+
+        # Identity/Connection String schenanigans
+        # If Identity and Connection String args are provided, Identity wins
+        if identity:
+            if endpoint_type.lower() == EndpointType.CosmosDBContainer.value:
+                if original_endpoint.primary_key or original_endpoint.secondary_key:
+                    logger.warning(NULL_WARNING.format("Primary and secondary keys"))
+                original_endpoint.primary_key = None
+                original_endpoint.secondary_key = None
+            else:
+                if original_endpoint.connection_string:
+                    logger.warning(NULL_WARNING.format("The connection string"))
+                original_endpoint.connection_string = None
+            original_endpoint.authentication_type = AuthenticationType.IdentityBased.value
+            if identity == SYSTEM_ASSIGNED_IDENTITY:
+                original_endpoint.identity = None
+            else:
+                original_endpoint.identity = ManagedIdentity(
+                    user_assigned_identity=identity
+                )
+        elif any([connection_string, primary_key, secondary_key]):
+            if original_endpoint.identity:
+                logger.warning(NULL_WARNING.format("The managed identity argument"))
```

Comment:
Would this be the identity `argument` or `property`?

Reply:
should be property